pkg/scheduler/plugins/gpushare/plugin.go

```diff
@@ -0,0 +1,295 @@
+		return framework.NewStatus(framework.Unschedulable, ErrNoMatchNode)
+	}
+
+	podRequest := state.convertedResource
+
+	var activeMinor int
+	for i := range *nodeDeviceTotal {
+		activeMinor = i
+		break
+	}
+
+	if gpuMem, ok := podRequest[apiext.GPUMemory]; ok {
+		podRequest[apiext.GPUMemoryRatio] = memBytesToRatio(gpuMem, (*nodeDeviceTotal)[activeMinor][apiext.GPUMemory])
+	} else {
+		gpuMemRatio := podRequest[apiext.GPUMemoryRatio]
+		podRequest[apiext.GPUMemory] = memRatioToBytes(gpuMemRatio, (*nodeDeviceTotal)[activeMinor][apiext.GPUMemory])
+	}
+
+	allocateResult := nodeDeviceInfo.tryAllocateGPU(podRequest)
+
+	g.nodeDeviceCache.assume(allocateResult, nodeName)
+
+	state.allocResult = allocateResult
+	return nil
+}
+
+func (g *Plugin) Unreserve(ctx context.Context, cycleState *framework.CycleState, pod *corev1.Pod, nodeName string) {
+	state, err := getPreFilterState(cycleState)
+	if err != nil {
+		return
+	}
+	if state.skip {
+		return
+	}
+	nodeDeviceInfo := g.nodeDeviceCache.getNodeDevice(nodeName)
+	if nodeDeviceInfo == nil {
+		return
+	}
+
+	g.nodeDeviceCache.unAssume(state.allocResult, nodeName)
+
+	state.allocResult = nil
+}
+
+func (g *Plugin) PreBind(ctx context.Context, cycleState *framework.CycleState, pod *corev1.Pod, nodeName string) *framework.Status {
+	state, err := getPreFilterState(cycleState)
+	if err != nil {
+		return framework.AsStatus(err)
+	}
+	if state.skip {
+		return nil
+	}
+
+	allocResult := state.allocResult
+	newPod := pod.DeepCopy()
+	if err := apiext.SetDeviceAllocations(newPod, schedulingv1alpha1.GPU, allocResult); err != nil {
+		return framework.NewStatus(framework.Error, err.Error())
+	}
+	patchBytes, err := util.GeneratePodPatch(pod, newPod)
+	if err != nil {
+		return framework.NewStatus(framework.Error, err.Error())
+	}
+	err = retryOnConflictOrTooManyRequests(func() error {
+		_, podErr := g.handle.ClientSet().CoreV1().Pods(pod.Namespace).
+			Patch(ctx, pod.Name, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{})
+		return podErr
+	})
+	if err != nil {
+		return framework.NewStatus(framework.Error, err.Error())
+	}
+
+	return nil
+}
+
+func New(args runtime.Object, handle framework.Handle) (framework.Plugin, error) {
+	extendedHandle, ok := handle.(frameworkext.ExtendedHandle)
+	if !ok {
+		return nil, fmt.Errorf("expect handle to be type frameworkext.ExtendedHandle, got %T", handle)
+	}
+
+	deviceCache := newNodeDeviceCache()
+
+	deviceInformerFactory := extendedHandle.KoordinatorSharedInformerFactory()
+	deviceInformer := deviceInformerFactory.Scheduling().V1alpha1().Devices().Informer()
+	deviceInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    deviceCache.onDeviceAdd,
+		UpdateFunc: deviceCache.onDeviceUpdate,
+		DeleteFunc: deviceCache.onDeviceDelete,
+	})
+	// make sure Device resources are loaded before Pods
+	deviceInformerFactory.Start(context.TODO().Done())
+	deviceInformerFactory.WaitForCacheSync(context.TODO().Done())
```

Comment:
you must recover all node account before onEvent execute, otherwise if you not load all used then begin to assign,  it will unexpected.